### ISSUE TITLE
add documentation

### ISSUE THREAD

**Issue by pavelnikonorov** [elixir-cloud-aai/ga4gh-sdk#16]
> Something to generate API docs from. It's very hard to read the code otherwise. Besides, we can use it to define doctests, as I understand.

_Originally posted by @uniqueg in https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/13#pullrequestreview-2155118075_

**Comment by aaravm** [comment]
While adding documentation in the code, I figure we could also address this along with it: https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/13#discussion_r1663330492

**Comment by uniqueg** [comment]
I also agree that providing documentation should be done continuously, with every proposed code change (as well as tests).
